docs/tutorials/user-management.txt

```diff
@@ -0,0 +1,181 @@
+.. _user-management:
+
+===============
+User Management
+===============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+The Mongo Ruby Driver provides a set of methods for managing users in a MongoDB deployment.
+All of these methods are defined on the ``Mongo::Auth::User::View`` class, which defines the
+behavior for performing user-related operations on a database. You can access a database's
+user view by calling the ``users`` method on the correpsonding ``Mongo::Database`` object:
+
+.. code-block:: ruby
+
+  client.database.users
+
+Note that this will open a view on the database to which the client is already connected.
+To interact with the users defined on a different database, call the client's ``use`` method
+and pass in the name of the database with which you want to connect:
+
+.. code-block:: ruby
+
+  client.use(:users).database.users
+
+In this example, all operations would be performed on the ``users`` database.
+
+For more information about users and user management, see MongoDB's
+:manual:`online documentation </core/security-users>`.
+
+Creating a user
+```````````````
+There are two ways to create a new database user with the Ruby Driver.
+
+The simplest way to create a new user is to use the ``create`` method, passing in a username,
+password, and roles:
+
+.. code-block:: ruby
+
+  client.database.users.create(
+    'alanturing',
+    password: 'enigma',
+    roles: [ Mongo::Auth::Roles::READWRITE ]
+  )
+
+Another way to create a user is to first create a ``Mongo::Auth::User`` object with all the
+user information and then pass that object into the ``create`` method instead.
+
+.. code-block:: ruby
+
+  user = Mongo::User.new({
+    user: 'alanturing',
+    password: 'enigma',
+    roles: [ Mongo::Auth::Roles::READWRITE ]
+  })
+
+  client.database.users.create(user)
+
+The ``create`` method takes a ``Hash`` of options as an optional second argument.
+The ``:roles`` option allows you to grant permissions to the new user.
+For example, the ``Mongo::Auth::Roles::READ_WRITE`` role grants the user the ability to both
+read from and write to the database in which they were created. Each role can be specified
+as a ``String`` or as a ``Hash``. If you would like to grant permissions to a user on a database
```

Comment:
I would love also for there to be a clarification (not necessarily made as part of this PR) as to what the auth source is for the created user. I.e. if I create a user on `foo` database's view, does the user have rights to `foo` database with an auth source of `admin` or is the user created with auth source of `foo`?

Reply:
Oh that's a good point. I'm still waiting to get the AWS access I need to deploy, so I'll just add it to this PR while I wait.

Reply:
I lied. I'll merge this and write up some information about auth source in a follow-up PR.